gcloud/datastore/key.py

```diff
@@ -83,9 +83,8 @@ def _validate_dataset_id(self):
                  can be implied.
         """
         if self._dataset_id is None:
-            if _implicit_environ.DATASET is not None:
-                # This assumes DATASET.id() is not None.
-                self._dataset_id = _implicit_environ.DATASET.id()
+            if _implicit_environ.DATASET_ID is not None:
```

Comment:
SGTM. My main goal here was to erase all evidence of `Dataset` and not much else.

Mind handling in the cherry-pick of #499?